core/src/main/java/org/mobilitydata/gtfsvalidator/parsing/RowParser.java

```diff
@@ -107,12 +110,16 @@ public boolean checkRowLength() {
   }
 
   @Nullable
-  public String asString(int columnIndex, boolean required) {
+  public String asString(int columnIndex, FieldLevelEnum level) {
     String s = row.asString(columnIndex);
-    if (required && s == null) {
+    if (level == FieldLevelEnum.REQUIRED && s == null) {
       noticeContainer.addValidationNotice(
           new MissingRequiredFieldNotice(
               fileName, row.getRowNumber(), header.getColumnName(columnIndex)));
+    } else if (level == FieldLevelEnum.RECOMMENDED && s == null) {
```

Comment:
If possible, this could also go in another PR specifically for the notices.

Reply:
I believe we will need to remove the `else if` statement here as the `MissingRecommendedFieldNotice` is not yet implemented in this PR. It should be in #1157.

Reply:
Actually, most of the checks failing in this PR are because of this - we are importing a class that does not exist in the import section.